streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -360,14 +333,8 @@ public void shouldRecycleStoreAndReregisterChangelog() {
         assertFalse(changelogReader.isPartitionRegistered(persistentStorePartition));
         assertThat(stateMgr.getStore(persistentStoreName), equalTo(store));
 
-        reset(context, store);
-        context.uninitialize();
```

Comment:
So counts are absolute which means that modifying `verify(context).uninitialize();` to `verify(context, times(2)).uninitialize()` is not necessary.  I did add `verify(context).uninitialize();` on line 336.

Reply:
The pre-migration code verfies that `context.uninitialize()` is called during the second `stateMgr.registerStateStores(singletonList(store), context);`. That means `verify(context).uninitialize();`  should be after the second `stateMgr.registerStateStores(singletonList(store), context);`. There you need to use `verify(context, times(2)).uninitialize()` since the test fails otherwise. The test should look like this:

```
    public void shouldRecycleStoreAndReregisterChangelog() {
        final ProcessorStateManager stateMgr = getStateManager(Task.TaskType.ACTIVE);
        final StateStore store = mock(StateStore.class);
        when(store.name()).thenReturn(persistentStoreName);
        stateMgr.registerStateStores(singletonList(store), context);

        verify(context).uninitialize();
        verify(store).init((StateStoreContext) context, store);

        stateMgr.registerStore(store, noopStateRestoreCallback, null);
        assertTrue(changelogReader.isPartitionRegistered(persistentStorePartition));

        stateMgr.recycle();
        assertFalse(changelogReader.isPartitionRegistered(persistentStorePartition));
        assertThat(stateMgr.getStore(persistentStoreName), equalTo(store));

        stateMgr.registerStateStores(singletonList(store), context);

        verify(context, times(2)).uninitialize();
        assertTrue(changelogReader.isPartitionRegistered(persistentStorePartition));
    }
```

Reply:
Maybe also here we misunderstood each other regarding the line numbers.

Reply:
That makes a lot more sense, just committed your change and pushed.